src/main/java/net/sacredlabyrinth/phaed/simpleclans/events/PlayerRankUpdateEvent.java

```diff
@@ -0,0 +1,63 @@
+package net.sacredlabyrinth.phaed.simpleclans.events;
+
+import net.sacredlabyrinth.phaed.simpleclans.Clan;
+import net.sacredlabyrinth.phaed.simpleclans.Rank;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.player.PlayerEvent;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * @author Minat0_
+ */
+public class PlayerRankUpdateEvent extends PlayerEvent implements Cancellable {
+
+    private static final HandlerList HANDLER_LIST = new HandlerList();
+    private final Rank oldRank;
+    private final Rank newRank;
+    private final Clan clan;
+    private boolean cancelled;
+
+    public PlayerRankUpdateEvent(@NotNull Player who, Clan clan, Rank oldRank, Rank newRank) {
```

Comment:
You forgot to update the constructor.

Reply:
That was RoinuijNosde memory, you don't understand it. 🤡